### PR TITLE
feat: add interview timestamp audit checks and clarify timestamp units

### PR DESCRIPTION
**Pull request by kaligrafy**
Add three new audit checks to validate interview timestamps:
- I_M_StartedAt: Validates that interview has a start timestamp
- I_I_StartedAtBeforeSurveyStartDate: Flags interviews started before survey period
- I_I_StartedAtAfterSurveyEndDate: Flags interviews started after survey period

The new checks respect survey configuration boundaries and handle edge cases (undefined timestamps, missing paradata, unconfigured survey dates).

Refactored timestamp handling to use explicit !== undefined checks instead of truthy checks, ensuring proper handling of edge cases like startedAt: 0. Added inline comments documenting the seconds-to-milliseconds conversion.

Updated InterviewParadata type documentation to explicitly specify that startedAt, updatedAt, and completedAt are stored as Unix epoch timestamps in seconds (not milliseconds) for clarity across the codebase.

Changes:
- Add InterviewAuditChecks: I_M_StartedAt, I_I_StartedAtBeforeSurveyStartDate, I_I_StartedAtAfterSurveyEndDate
- Add comprehensive test suites for all three audit checks (17 tests total)
- Add English and French translations for audit error messages
- Update InterviewParadata type definitions with clarified timestamp units
- Improve timestamp conversion logic for better readability and correctness

## Summary by CodeRabbit

* **New Features**
  * Added interview start-time audit validations (missing, before survey start, after survey end) and corresponding English/French messages.
  * Added utility functions to parse/convert timestamps (seconds ⇄ milliseconds).

* **Documentation**
  * Clarified that timestamp fields are unix epoch seconds (not milliseconds) in inline comments.

* **Tests**
  * Added comprehensive unit tests covering pass/fail cases, cross-timezone scenarios, and edge cases for the new checks and utilities.

### CODE REVIEW

File: packages/evolution-common/src/services/baseObjects/interview/InterviewParadata.ts [hunk 1]
```diff
@@ -33,9 +33,9 @@ const interviewParadataAttributes = [
 const interviewParadataAttributesWithComposedAttributes = [...interviewParadataAttributes];
 
 export type InterviewParadataAttributes = {
-    startedAt?: number; // unix epoch timestamp;
-    updatedAt?: number; // unix epoch timestamp;
-    completedAt?: number; // unix epoch timestamp; updated at the end of the interview. Needs auditing before assuming the interview is really completed.
+    startedAt?: number; // unix epoch timestamp in seconds (not milliseconds)
+    updatedAt?: number; // unix epoch timestamp in seconds (not milliseconds)
+    completedAt?: number; // unix epoch timestamp in seconds (not milliseconds); updated at the end of the interview. Needs auditing before assuming the interview is really completed.
```
[the remark this answers is not in the document]
Comment (kaligrafy) [reply packages/evolution-common/src/services/baseObjects/interview/InterviewParadata.ts]: should be fixed

File: packages/evolution-backend/src/services/audits/auditChecks/checks/__tests__/interview/I_I_StartedAtAfterSurveyEndDate.test.ts [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    const validUuid = uuidV4();
+
+    // Save original config value
+    const originalEndDate = projectConfig.endDateTimeWithTimezoneOffset;
+
+    afterEach(() => {
+        // Restore original config value
+        projectConfig.endDateTimeWithTimezoneOffset = originalEndDate;
+    });
+
```
[the remark this answers is not in the document]
Comment (kaligrafy) [reply packages/evolution-backend/src/services/audits/auditChecks/checks/__tests__/interview/I_I_StartedAtAfterSurveyEndDate.test.ts]: should be fixed, please verify
